### PR TITLE
Add clustering as a problem type

### LINKED ISSUE #2874: Add clustering as a problem type
We should also make sure users get useful errors if they try to use AutoMLSearch or other functions irrelevant to clustering problem types.

### PR DESCRIPTION
**Pull request by eccabay**
Closes #2874

### CODE REVIEW

**Review by bchen1116 — APPROVED** [review]
LGTM! 
This was primarily to add clustering as a problem type, right? The issue itself has `Also within scope of this issue, add the first clustering objective(s) and algorithm(s).` in the description, but there's no objectives or algos added?

**Comment by eccabay** [comment]
@bchen1116 yup, great point! This issue is just for adding clustering as a problem type, the issue is outdated. There are separate issues for adding clustering models and objectives, so I'll update this one to be more accurate.

**Review by chukarsten — APPROVED** [review]
Looks good!  Sorry the review took so long!

File: evalml/automl/automl_search.py [hunk 1]
```diff
@@ -475,6 +476,8 @@ def __init__(
                 "Time series support in evalml is still in beta, which means we are still actively building "
                 "its core features. Please be mindful of that when running search()."
             )
+        if is_clustering(self.problem_type):
```
[the remark this answers is not in the document]
Comment (eccabay) [reply evalml/automl/automl_search.py:479]: It doesn't seem like it, at least for the class docstring. `AutoMLSearch.__init__` throws a bunch of `ValueError`s that aren't listed.

**Pull request merged**